File: actionlib_tutorials/scripts/fibonacci_client.py
import rospy
from __future__ import print_function

import actionlib

import actionlib_tutorials.msg

def fibonacci_client():

    client = actionlib.SimpleActionClient('fibonacci',actionlib_tutorials.msg.FibonacciAction)
    client.wait_for_server()
    goal  =actionlib_tutorials.msg.FibonacciGoal(order=20)
    client.send_goal(goal)
    client.wait_for_result()
    return client.get_result()

if __name__ == "__main__":
    try:
        rospy.init_node('fibonacci_client_py')
        result = fibonacci_client()
        print("Result:",', '.join([str(n) for n in result.sequence]))
    except rospy.ROSInterruptException:
        print("program interrupted before completion",file = sys.stderr)

File: actionlib_tutorials/src/fibonacci_client.cpp
#include <ros/ros.h>
#include <actionlib/client/simple_action_client.h>
#include <actionlib/client/terminal_state.h>
#include <actionlib_tutorials/FibonacciAction.h>

int main(int argc , char **argv){
    ros::init(argc, argv, "test_fibonacci");

    actionlib::SimpleActionClient<actionlib_tutorials::FibonacciAction> ac("fibonacci",true);

    ROS_INFO("Waiting for action server to start.");

    ac.waitForServer();

    ROS_INFO("Action Server started , sending goal.");

    actionlib_tutorials::FibonacciGoal goal;
    goal.order = 20;
    ac.sendGoal(goal);

    bool finished_before_timeout = ac.waitForResult(ros::Duration(30.0));

    if (finished_before_timeout){
        actionlib::SimpleClientGoalState state = ac.getState();
        ROS_INFO("Action finished: %s", state.toString().c_str());
    }
    else
        ROS_INFO("Action did not finish before the time out.");

    return 0;
}
